### PR TITLE
endpoint: Don't fail with fatal on l4 policy application

### LINKED ISSUE #3197: Cilium died when policy was applied
Happens on Runtime Policies L7: 

The cilium log is the following:
```
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="consumable regenerated" containerID=3699b04bbf egressSecurityIdentities="map[]" endpointID=22475 identity=55657 ingressSecurityIdentities="map[]" l4Rm="map[]" policyRevision=131 rulesAdd="map[]" rulesRm="map[]"
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="Done regenerating" containerID=3699b04bbf endpointID=22475 optsChanged=false policyChanged=true policyRevision=131 policyRevision.next=132
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="TriggerPolicyUpdatesLocked: changed: true" containerID=3699b04bbf endpointID=22475 policyRevision=131
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="Starting regenerate..." containerID=78b21f364e endpointID=35325 policyRevision=131
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="No change in CIDR policy" containerID=78b21f364e endpointID=35325 policyRevision=131
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="Policy Ingress enabled" containerID=78b21f364e endpointID=35325 policyRevision=131
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="L4 Policy matches" PolicyID.L4=host identityLabels="[reserved:host]"
Mar 19 15:00:57 runtime cilium-agent[9298]: level=warning msg="Update of l4 policy map failed" containerID=78b21f364e endpointID=35325 error="Unable to update element: invalid argument" policyID=1 policyRevision=131 port=80 protocol=6
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="L4 Policy matches" PolicyID.L4=52632 identityLabels="[container:id.app1]"
Mar 19 15:00:57 runtime cilium-agent[9298]: level=warning msg="Update of l4 policy map failed" containerID=78b21f364e endpointID=35325 error="Unable to update element: invalid argument" policyID=52632 policyRevision=131 port=80 protocol=6
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="L4 Policy matches" PolicyID.L4=host identityLabels="[reserved:host]"
Mar 19 15:00:57 runtime cilium-agent[9298]: level=warning msg="Update of l4 policy map failed" containerID=78b21f364e endpointID=35325 error="Unable to update element: invalid argument" policyID=1 policyRevision=131 port=8080 protocol=6
Mar 19 15:00:57 runtime cilium-agent[9298]: level=debug msg="L4 Policy matches" PolicyID.L4=52632 identityLabels="[container:id.app1]"
Mar 19 15:00:57 runtime cilium-agent[9298]: level=warning msg="Update of l4 policy map failed" containerID=78b21f364e endpointID=35325 error="Unable to update element: invalid argument" policyID=52632 policyRevision=131 port=8080 protocol=6
Mar 19 15:00:57 runtime cilium-agent[9298]: level=fatal msg="L4 Policy application failed" containerID=78b21f364e endpointID=35325 error="Some Label+L4 policy updates failed." policyRevision=131
Mar 19 15:00:57 runtime cilium-agent[9298]: level=panic msg="Agent pipe closed, shutting down"
```

[RuntimeValidatedPoliciesL7Checks.zip](https://github.com/cilium/cilium/files/1825627/RuntimeValidatedPoliciesL7Checks.zip)

https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/1765/

All data attached.

### PR DESCRIPTION
**Pull request by tgraf**
A better fix should be put in place via #3164 

Fixes: #3197

Signed-off-by: Thomas Graf <thomas@cilium.io>

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/endpoint/policy.go [hunk 1]
```diff
@@ -362,7 +362,7 @@ func (e *Endpoint) regenerateConsumable(owner Owner, labelsMap *identityPkg.Iden
 			rulesAdd, l4Rm, err = e.applyL4PolicyLocked(owner, labelsMap, e.L4Policy, c.L4Policy)
 			if err != nil {
 				// This should not happen, and we can't fail at this stage anyway.
-				e.getLogger().WithError(err).Fatal("L4 Policy application failed")
```
Comment (raybejjani) [inline pkg/endpoint/policy.go:365]: Should you return here?

Comment (manalibhutiyani) [reply pkg/endpoint/policy.go:365]: +1 ^ Should be return here.

Comment (tgraf) [reply pkg/endpoint/policy.go:365]: And not enforce the policy at all?

Comment (manalibhutiyani) [reply pkg/endpoint/policy.go:365]: Wouldn't the policy enforced after this failure only be partially correct i.e some Label+L4 policy updates would have possibly failed.? Curious what's a better trade-off, to enforce partial policy or complain and not allow any policy.

Comment (tgraf) [reply pkg/endpoint/policy.go:365]: Sorry, overlooked something. Yes, we can still fail.

**Comment by tgraf** [comment]
test-me-please

**Review by manalibhutiyani — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**